lib/stack_master/parameter_resolvers/one_password.rb

```diff
@@ -0,0 +1,49 @@
+module StackMaster
+  module ParameterResolvers
+    class OnePassword < Resolver
+      OnePasswordNotFound = Class.new(StandardError)
+
+      array_resolver
+
+      def initialize(config, stack_definition)
+        @config = config
+        @stack_definition = stack_definition
+      end
+
+      def resolve(params={})
+        raise RuntimeError, "1password requires the `OP_SESSION_<name>` to be set" if ENV.keys.grep(/OP_SESSION_\w+$/).empty?    
+        get_items(params)
+      end
+
+      private
+
+      def op_get_item(item, vault)
+        begin
+          item = %x(op get item --vault='#{vault}' '#{item}' 2>&1)
+          return item if JSON.parse(item)
+        rescue Errno::ENOENT
+          raise RuntimeError, "The op cli needs to be installed in the PATH"
+        rescue => exception
+          raise RuntimeError, "Failed to return item from 1password, #{item}"
+        end
+      end
+
+      def get_password(title, vault)
+        JSON.parse(op_get_item(title, vault))['details']['fields'].select { |k,v| k.key('password') }[0].values_at('value').first
```

Comment:
I think this could be wrapped up a bit better to make it easier to grok and deal with. Should the results from `op_get_item` be an OpenStruct or a Class? This would also enable us to test against multiple versions of the `op` client version if we need to.

Reply:
I toyed with `OpenStruct` previously. I didn't realise I could add it as a output to `JSON.parse`! This will change my world in so many ways.